### PR TITLE
worked with third party api and created emailservice

### PR DESCRIPTION
**Pull request by Tokhirbek7**

### CODE REVIEW

File: src/main/java/com/vention/stock_market_share/controller/EmailSendController.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+@RestController
+@RequestMapping("/mail")
+public class EmailSendController {
+    private final EmailService emailService;
+
+    public EmailSendController(EmailService emailService) {
+        this.emailService = emailService;
+    }
+
+    @PostMapping()
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/controller/EmailSendController.java:19]: For what we use this controller?

File: src/main/java/com/vention/stock_market_share/controller/StockController.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import com.vention.stock_market_share.response.AlphaVintageResponse;
+import com.vention.stock_market_share.service.AlphaVintageService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/controller/StockController.java:11]: use @RequestMapping("/stock")
and delete ("/searchStockBySymbol") and ("/searchAllTradingShares")

For All just @GetMapping, for one use @GetMapping("/{symbol}") like PathVariable

File: src/main/java/com/vention/stock_market_share/interfaces/AlphaVintageClient.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.vention.stock_market_share.interfaces;
+
+import com.vention.stock_market_share.response.AlphaVintageResponse;
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+@Component
+@FeignClient(name = "AlphaVintageClient", url = " ${alphavantage.base.url}")
+public interface AlphaVintageClient {
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/interfaces/AlphaVintageClient.java:10]: https://github.com/Poviraevsergei/StockMarket/blob/master/src/main/java/by/itechart/service/serviceFeign/StockServiceFeign.java

use variant like this, don't use body. just pathVariable here. And move apiKey ti application.property file

File: src/main/java/com/vention/stock_market_share/response/AlphaVintageResponse.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
+package com.vention.stock_market_share.response;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Data;
+
+@Data
+public class AlphaVintageResponse {
+    @JsonProperty("symbol")
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/response/AlphaVintageResponse.java:8]: let's use Path variable

File: src/main/java/com/vention/stock_market_share/service/EmailService.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            mimeMessageHelper.setText(body);
+            for (int i = 0; i < file.length; i++) {
+                mimeMessageHelper.addAttachment(
+                        file[i].getOriginalFilename(),
+                        new ByteArrayResource(file[i].getBytes())
+                );
+            }
+            mailSender.send(mimeMessage);
+            return "mail sent";
+
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/service/EmailService.java:56]: why empty line

File: src/main/java/com/vention/stock_market_share/service/EmailService.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        } catch (Exception e) {
+            throw new RuntimeException();
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/service/EmailService.java:58]: do not throw exception in catch block. Just log it here.

**Review by Poviraevsergei — CHANGES_REQUESTED** [review]

File: src/main/java/com/vention/stock_market_share/controller/EmailSendController.java (hunk 1 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/controller/EmailSendController.java:19]: I just tried how to send emails with multipart files. Now i deleted EmailController

File: src/main/java/com/vention/stock_market_share/controller/StockController.java (hunk 2 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/controller/StockController.java:11]: Done

File: src/main/java/com/vention/stock_market_share/service/EmailService.java (hunk 5 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/service/EmailService.java:56]: Fixed the problem

File: src/main/java/com/vention/stock_market_share/service/EmailService.java (hunk 6 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/service/EmailService.java:58]: Done

File: src/main/java/com/vention/stock_market_share/interfaces/AlphaVintageClient.java (hunk 3 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/interfaces/AlphaVintageClient.java:10]: Done

**Pull request merged**